Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,144 @@
+is not suitable for PCI devices because it utilizes the platform bus that is
+meant for platform devices that are controlled by DT.
+
+Virtual Bus Devices
+~~~~~~~~~~~~~~~~~~~
+
+A virtual bus device represents an individual function of its parent PCI device.
+It is given a match_name that is used for driver binding and a release callback
+that is invoked when the device is unregistered.
+
+.. code-block:: c
+
+	struct virtbus_device {
+		struct device dev;
+		const char *match_name;
+		void (*release)(struct virtbus_device *);
+		u32 id;
+	};
+
+Virtual bus devices can be used in a lot of subsystems and they need not have
+anything in common with other devices on the virtual bus. The PCI device that
+needs these devices is responsible for registering/unregistering and
+maintaining its child virtual bus devices.
+
+Virtual Bus Drivers
+~~~~~~~~~~~~~~~~~~~
+
+Virtual bus drivers follow the standard driver model convention, where
+discovery/enumeration is handled outside the drivers, and drivers
+provide probe() and remove() methods. They support power management
+and shutdown notifications using the standard conventions.
+
+.. code-block:: c
+
+	struct virtbus_driver {
+		int (*probe)(struct virtbus_device *);
+		int (*remove)(struct virtbus_device *);
+		void (*shutdown)(struct virtbus_device *);
+		int (*suspend)(struct virtbus_device *, pm_message_t);
+		int (*resume)(struct virtbus_device *);
+		struct device_driver driver;
+		const struct virtbus_device_id *id_table;
+	};
+
+Virtual bus drivers register themselves by calling virtbus_register_driver().
+The id_table contains the names of virtbus devices that can bind with the
+driver.
+
+Device Enumeration
+~~~~~~~~~~~~~~~~~~
+
+The virtbus device is enumerated when it is attached to the bus. The device
+is assigned a unique ID automatically that will be appended to its name. If
+two virtbus_devices both named "foo" are registered onto the bus, they will
+have the device names, "foo.x" and "foo.y", where x and y are unique integers.
+
+Driving Binding
+~~~~~~~~~~~~~~~
+
+When a virtual bus device is registered, the driver core iterates through the
+drivers registered with the virtual bus to identify a match between the device's
+match_name and the names in the driver's id_table. Once a successful match is
+found, the driver's probe is invoked binding the device to the driver upon
+success.
+
+Example Usage
+~~~~~~~~~~~~~
+
+Virtual bus devices are created and registered by a multi-function PCI device.
+Therefore, they are typically encapsulated within a structure defined by
+parent PCI device which contains the virtual bus device and any associated
+shared data/callbacks needed to establish the connection with the parent.
+
+An example would be:
+
+.. code-block:: c
+
+        struct foo {
+                struct virtbus_device vdev;
+		void (*connect)(struct virtbus_device *vdev);
+		void (*disconnect)(struct virtbus_device *vdev);
+                void *data;
```

Comment:
It is blank instead of tab in the front of  ``struct virtbus_device vdev;`` and ``void *data;``